core/templates/dev/head/pages/signup/Signup.js

```diff
@@ -173,9 +173,44 @@ oppia.controller('Signup', [
       $http.post(_SIGNUP_DATA_URL, requestParams).then(function() {
         window.location = window.decodeURIComponent(
           UrlService.getUrlParams().return_url);
-      }, function() {
+      }, function(rejection) {
+        if (
+          rejection.data && rejection.data.status_code === 401) {
+          $scope.showContinueRegistrationModal();
+        }
         $scope.submissionInProcess = false;
       });
     };
+
+    $scope.showContinueRegistrationModal = function() {
+      $uibModal.open({
+        templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+          '/pages/signup/redirect_to_login_directive.html'),
+        backdrop: 'static',
+        keyboard: false,
+        resolve: {},
+        controller: [
+          '$scope', '$uibModalInstance', 'SiteAnalyticsService',
+          'UserService', '$timeout', '$window',
+          function($scope, $uibModalInstance, SiteAnalyticsService,
+              UserService, $timeout, $window) {
+            $scope.continueRegistration = function() {
+              SiteAnalyticsService.registerStartLoginEvent(
```

Comment:
I don't think this really counts as a second login event, so let's delete this.

Reply:
Done.